jmclient/jmclient/maker.py

```diff
@@ -171,8 +171,7 @@ def on_tx_received(self, nick, tx_from_taker, offerinfo):
                 # unchanged as (sig, pub).
                 try:
                     sig, pub = [a for a in iter(tx.wit.vtxinwit[index].scriptWitness)]
-                    scriptCode = btc.pubkey_to_p2wpkh_script(pub)
-                    sigmsg = btc.CScript([sig]) + btc.CScript(pub) + scriptCode
+                    sigmsg = btc.CScript([sig]) + btc.CScript(pub)
```

Comment:
Just stating the relatively obvious here: this means new code cannot do Joinmarket coinjoins with p2sh. I think it's the right decision, but in principle, we could also do a check on the wallet type and then change the sig msg to be one of these two syntaxes.
The reason I think it's the right choice is (a) keeping logic simpler and (b) if we have an upgrade like this, we do want to strongly encourage people to shift over. There are counterarguments though, I don't know if anyone wants to make that.

Given that decision though, this is the right code change, of course.

Reply:
In any case I plan to continue running P2SH yg for some time (as I previously did with P2PKH), instead of sweeping everything to a new native segwit bech32 wallet, but maybe it's ok to continue using v0.6.x for that. Downside - unability to do P2SH BIP78 payjoins from such wallet in case that is what receiving side is supporting.

Reply:
> Just stating the relatively obvious here: this means new code cannot do Joinmarket coinjoins with p2sh.

@AdamISZ I'm assuming that you mean that p2sh is nested p2wkh. It seems to me that those changes, in conjunction with those on the taker side, should still support p2sh. What I've done is I've brought the creation of `scriptCode` entirely on the taker side. These changes depend entirely on the assumption that p2sh is np2wkh as far as coinjoins are concerned. I'd encourage people to take a look at the `taker.py` changes to understand my reasoning here.

> Downside - unability to do P2SH BIP78 payjoins from such wallet in case that is what receiving side is supporting.

@kristapsk If my reasoning is correct above, then I think P2SH BIP78 payjoins will still be possible with these changes?

Reply:
> I think it's the right decision, but in principle, we could also do a check on the wallet type and then change the sig msg to be one of these two syntaxes.

I actually think this would improve the code here and make it cleaner! I'm thinking of a simple if statement conditioned on the wallet type to set the correct syntax for `sigmsg`. This assumes that all the inputs from a single maker will be of the same type. If people like this, I'll make this change

Reply:
> Downside - unability to do P2SH BIP78 payjoins from such wallet in case that is what receiving side is supporting.

Well, first, I think we should release 0.7.0 without this PR and second you could still do BIP78 payjoins with this PR included, just as today you can do BIP78 payjoins with native wallets.

Reply:
> > Just stating the relatively obvious here: this means new code cannot do Joinmarket coinjoins with p2sh.
> 
> @AdamISZ I'm assuming that you mean that p2sh is nested p2wkh. It seems to me that those changes, in conjunction with those on the taker side, should still support p2sh. What I've done is I've brought the creation of `scriptCode` entirely on the taker side. These changes depend entirely on the assumption that p2sh is np2wkh as far as coinjoins are concerned. I'd encourage people to take a look at the `taker.py` changes to understand my reasoning here.
> 

I wrote this comment before reading how you had changed `Taker.on_sig()` (see my extended comment on that below). It was not correct, so you can ignore this comment.

Reply:
> I actually think this would improve the code here and make it cleaner! I'm thinking of a simple if statement conditioned on the wallet type to set the correct syntax for `sigmsg`. This assumes that all the inputs from a single maker will be of the same type. If people like this, I'll make this change.

So as per above, I agree this is probably the best. It doesn't hurt to add this functionality, even if we are defaulting new makers to bech32 (after this PR).

Reply:
> you could still do BIP78 payjoins with this PR included

My comment was about v0.6.x, where BIP78 payjoin support is not included.